### DISCUSSION TITLE
Store themes on cookies

### DISCUSSION THREAD

**Discussion by washington299** [vercel/next.js#11577] [Help]
Can someone tell me how to switch light mode and dark mode themes with styled-components and persist it on cookies, I’m trying to do this for my portfolio but it’s not working, and I can’t fix it, It would be great an example like this inside the examples folder.

**Comment by Dorson** [comment]
I Coded the dark mode switch in JS and basic CSS.

My version does store the color preference in localStorage of the browser, for the speed. Cookies are server-side. In localStorage we can have settings without logins or user profiles.

 In my version the multiple color themes can be defined inside of one CSS style file. Many color theme in one file :-)

Try it on the Github demo page :
https://dorson.github.io/CSS-Dark-Mode-and-color-switch/

Feel free to fork, copy, adopt, etc...

Have fun ! :+1:

**Reply by washington299** [reply]
Thanks for your reply man, I got the solution setting the cookies on the _app file and it persists normally, you can see that in my portfolio if you want :) https://portfolio.washingtoncampos9.now.sh/

**Reply by Dorson** [reply]
Great Job !
